scripts/extract_pdf_abstract.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+# coding: utf8
+"""Extracts the abstracts from the proceedings PDFs.
+
+Usage
+-----
+
+$ python ./scripts/extrac_pdf_abstract.py \
+    ./path/to/pdfs
+
+"""
+import os
+import glob
+import io
+import tempfile
+import argparse
+import pdfminer.high_level
+import pdfminer.layout
+import pdfminer.settings
+from pdfrw import PdfReader, PdfWriter
+from pdfrw.findobjs import page_per_xobj
+pdfminer.settings.STRICT = False
+
+
+def extract_first_page(fname):
+
+    # create a temporary PDF file
+    path_tmpfile = os.path.join(tempfile.gettempdir(), os.path.basename(fname))
+
+    # extract all elements from the PDF and put them on separate pages
+    pages = list(page_per_xobj(PdfReader(fname).pages, margin=0.5*72))
+
+    # write to temp file
+    writer = PdfWriter(path_tmpfile)
+    writer.addpages(pages)
+    writer.write()
+
+    return path_tmpfile
+
+
+def extract_text(fname):
+    """Get all the text from the PDF."""
+
+    laparams = pdfminer.layout.LAParams()
+    for param in ('all_texts', 'detect_vertical', 'word_margin', 'char_margin', 'line_margin', 'boxes_flow'):
+        paramv = locals().get(param, None)
+        if paramv is not None:
+            setattr(laparams, param, paramv)
+
+    # send output to a string stream
+    outfp = io.StringIO()
+
+    with open(fname, 'rb') as fp:
+        pdfminer.high_level.extract_text_to_fp(fp, outfp=outfp, codec='utf-8',
+                                               laparams=laparams, pages=0)
+
+    return outfp.getvalue()
+
+
+def extract_abstract(raw_text):
+    """Search in the text for keywords and extract text in between."""
+
+    abs_index = raw_text.find('ABSTRACT')
+    intro_index = abs_index
+
+    intro_index = raw_text.find('1. INTRODUCTION')
```

Comment:
Haha, yes indeed, this would make it easier.

I think there will be some kind of config for each conference year, since sometimes they are embedded, sometimes not. I'll add this on the go then...

Reply:
Ah, but the indexing will shift so the position needs to be mapped back then.
I will just leave it as is but put some assertions in there so it doesn't crash and gives some warning.

Reply:
ya, that makes sense